core/domain/collection_domain.py

```diff
@@ -230,13 +263,54 @@ def create_default_node(cls, exploration_id):
         return cls(exploration_id)
 
 
+class CollectionDict(TypedDict, total=False):
```

Comment:
Total is added because this Dict is loose in terms of having exact number of keys.
for eg:- `to_dict` function of this class returning Dict representaion of collection object but without `id` key.

Also, `total` is used when we want to exclude a dict key from a given `typeddict class`, but we cannot add extra key to this typeddict class. All keys which are subset of this typeddict class should follow exact annotations.

Reply:
So the only non-total element is `id`?

Reply:
No, because here we deleting the id key also!
https://github.com/oppia/oppia/blob/add62e844665ab43229dbfd3cd207dc0e0343356/core/domain/collection_domain.py#L426
Also, when we using to_dict() method `created_on` and `last_updated` keys is not returned. So, there is total 3 loose keys.